examples/tree_2d_dgsem/elixir_euleracoustics_co-rotating_vortex_pair.jl

```diff
@@ -0,0 +1,347 @@
+
+  return prim2cons(s, equations)
+end
+
+function calc_damping_factor(x, direction, sponge_layer_min, sponge_layer_max)
+  # Damping factor alpha grows linearly from 0 to 1 depending on how deep x lies in the sponge layer
+  # If x does not lie in the sponge layer, this returns 0
+
+  # Get the coordinate that determines how deep we are in the sponge zone
+  if direction in (1, 2)
+    pos = x[1]
+  else
+    pos = x[2]
+  end
+
+  # Determine where the sponge layer begins/ends to allow calculating the damping factor
+  if iseven(direction)
+    sponge_begin = sponge_layer_min[direction]
+    sponge_end = sponge_layer_max[direction]
+  else
+    sponge_begin = sponge_layer_max[direction]
+    sponge_end = sponge_layer_min[direction]
+  end
+
+  alpha = (pos - sponge_begin) / (sponge_end - sponge_begin)
+
+  # alpha lies in [0, 1] if and only if x lies in the sponge zone
+  if 0.0 <= alpha <= 1.0
+    return alpha
+  else
+    return 0.0
+  end
+end
+
+
+# Boundary condition for the flow problem: The sponge layer dampens density and pressure towards the
+# freestream values. The freestream values (converted into conservative variables) are therefore
+# used as a Dirichlet boundary
+struct BoundaryCondition{uEltype}
+  rho::uEltype
+  rho_e::uEltype
+end
+
+function (bc::BoundaryCondition)(u_inner, orientation, direction, x, t,
+                                 surface_flux_function,
+                                 equations::CompressibleEulerEquations2D)
+  uEltype = eltype(u_inner)
+  u_boundary = SVector(bc.rho, zero(uEltype), zero(uEltype), bc.rho_e)
+
+  # Calculate boundary flux
+  if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
+end
+
+end # module
+
+
+import .VortexPairSetup
+
+
+###############################################################################
+# shared parameters, mesh and solver for both semidiscretizations
+
+# Parameters of the vortex pair
+Mach = 1/9
+c0 = 1.0
+r0 = 1.0
+circulation = 4 * pi * r0 * c0 * Mach
+rho = 1.0
+
+rc = 2/9 * r0 * 1.0
+
+T_r = 8 * pi^2 * r0^2 / circulation # Rotational period of the vortex pair
+T_a = T_r / 2 # Acoustic period of the vortex pair
+
+vortex_pair = VortexPairSetup.VortexPair(r0, rc, c0, circulation, rho)
+
+# Shared mesh for both semidiscretizations
+coordinates_min = (-135*r0, -135*r0) # minimum coordinates (min(x), min(y))
+coordinates_max = ( 135*r0,  135*r0) # maximum coordinates (max(x), max(y))
+refinement_patches = (
+  (type="sphere", center=(0.0, 0.0), radius=85.0*r0),
+  (type="sphere", center=(0.0, 0.0), radius=20.0*r0),
+  (type="sphere", center=(0.0, 0.0), radius=10.0*r0),
+  (type="sphere", center=(0.0, 0.0), radius=5.0*r0)
+)
+initial_refinement_level=7
```

Comment:
Yes, IIRC this was necessary to be able to reproduce the reference values.

Reply:
Yes, fully resolving the acoustic pressure field for this setup requires a very high resolution. In fact, I used `initial_refinement_level=8` to reproduce the reference values because level 7 actually isn't sufficient (DG elements are still clearly visible on line plots, even though the spiral pattern only shows minor distortions). IIRC it took about 8 hours with 16 threads on Rocinante (with level 8)...

I could try to reduce the runtime by using a smaller domain and final time, so that people interested in this can actually run this elixir themselves with reasonable amount of effort. What do you think?

Reply:
Maybe it would be good to add this in a general comment at the beginning of the elixir? E.g., expand your one-line comment with a mini description of the setup, a reference to the relevant paper(s), and a warning that this takes a long time to run in full?

Reply:
Added a comment and references in c6a9c78.